tests/test_task.py

```diff
@@ -1365,22 +1365,25 @@ def test_return_value_no_record(self):
 
     def test_return_value(self):
         """TaskGraph: test that ``.get`` behavior works as expected."""
+        if hasattr(_return_value_once, 'executed'):
+            del _return_value_once.executed
         n_iterations = 3
+        task_graph = taskgraph.TaskGraph(self.workspace_dir, 0, 0)
         for iteration_id in range(n_iterations):
             transient_run = iteration_id == n_iterations-1
             LOGGER.debug(iteration_id)
-            task_graph = taskgraph.TaskGraph(self.workspace_dir, 0, 0)
             expected_value = 'a good value'
             value_task = task_graph.add_task(
                 func=_return_value_once,
                 transient_run=transient_run,

```

Comment:
This task does not re-execute on the last iteration when `transient_run=True`. The func would raise a `RuntimeError` if it did, right? Shouldn't `transient_run=True` always force an execution?

Reply:
Now this one IS intentional. Given my comment above, it doesn't make sense to add duplicate Tasks to TaskGraph and the way it's set up is that if you do it doesn't break anything and instead logs an error. So this bit here is just ensuring that multiple transient run tasks in the same TaskGraph instance are only executed once. I thought that behavior was worth explicitly testing since I do it with non transient tasks elsewhere in the suite.